Automattic/newspack-ios
# Issues/data fixes and new datasource

This is a multipurpose PR. It introduces a `UITableViewDiffableDataSource` to back the post list. It also addresses an annoying core data merge error that seems to be related to a complex entity relationship pattern that (according to [the docs](https://developer.apple.com/documentation/coredata/nsmergeconflict)) created a conflict between the persistent store layer, and the external store (database) layer. The issue seems to be present in `develop`, but was exacerbated by the introduction of the `UITableViewDiffableDataSource`. (It would crash 3/5 times). 

You should be able to confirm the merge error by logging out, then logging in and attempting to rapidly scroll the post list so that its loading the next page of post items while still loading individual rows. It might take a number of tries to get it to trigger (I didn't know it was there 😬 but could eventually reproduce it in develop). 

The crash is addressed by removing the site relationship on `PostItem` and `Post` and replacing it with the site's UUID.

To test: 
- Bonus: Confirm the crash in the `develop` branch.
- Switch to this branch.  Confirm tests pass.
- Launch the app and view the post list. 
- Confirm the post list loads as expected. 
- Rapidly scroll the list a lot. 
- Confirm new rows load as expected and there is no crash. 
- Confirm scrolling is reasonably snappy. 
- Repeat the post list test a few times after logging out and back in.  Confirm there is no crash.

@jleandroperez would you mind taking a peek at this one?
